api/py/ai/chronon/repo/compile.py

```diff
@@ -126,25 +131,133 @@ def extract_and_convert(chronon_root, input_path, output_root, debug, force_over
+        dependencies = {}
+        dependencies.update({**extra_dependent_group_bys_to_materialize, **extra_dependent_joins_to_materialize})
+        assert not dependencies, (
+            "You must also materialize all dependent GroupBys or Joins."
+            " You can do this by passing the --force-overwrite flag to the compile.py command."
+            " Detected dependencies are as follows: {}".format(sorted(dependencies.keys()))
+        )
+
+    if extra_online_group_bys or extra_dependent_group_bys_to_materialize:
+        extra_online_group_bys.update(extra_dependent_group_bys_to_materialize)
+        _handle_extra_conf_objects_to_materialize(
+            conf_objs=extra_online_group_bys,
+            force_overwrite=force_overwrite,
+            full_output_root=full_output_root,
+            log_level=log_level,
+            teams_path=teams_path,
+            validator=validator,
+        )
+    if extra_dependent_joins_to_materialize:
+        _handle_extra_conf_objects_to_materialize(
+            conf_objs=extra_dependent_joins_to_materialize,
+            force_overwrite=force_overwrite,
+            full_output_root=full_output_root,
+            log_level=log_level,
+            teams_path=teams_path,
+            validator=validator,
+            is_gb=False,
+        )
     if num_written_objs > 0:
         print(f"Successfully wrote {num_written_objs} {(obj_class).__name__} objects to {full_output_root}")
 
 
+def _get_relative_materialized_file_path(full_output_root, name, obj):
+    _, _, abs_output_file_path = _construct_output_file_name(full_output_root, name, obj)
+    output_file_path = abs_output_file_path.replace(full_output_root + os.sep, "")
+    return output_file_path
+
+
+# todo add type hints and param arguments to all functions and methods
+# todo why does force-overwrite not work for inline group_bys chaining. no warning printed. worked in prod
+def _get_conf_file_path(downstream, downstream_class):
+    parts = downstream.split(".")
+
+    # This may happen for scenarios where group_by is defined within join file itself.
+    # So no separate group_by file is present. In such cases, compiling the join will handle the materialize
+    # for group_by.
+    if len(parts) < 3:
+        return None
+
+    team_name = parts[0]
+    conf_var = parts[-1]
+    conf_name = parts[-2] + ".py"
+
+    # Nested path between team_name and conf_name
+    nested_path = parts[1:-2] if len(parts) > 3 else []
+
+    # Construct the path for configuration file
+    conf_file_path = os.path.join(team_name, *nested_path, conf_name)
+
+    # Return the appropriate path based on downstream_class
+    if downstream_class is Join:
+        return conf_var, os.path.join(JOIN_FOLDER_NAME, conf_file_path)
+    elif downstream_class is GroupBy:
+        return conf_var, os.path.join(GROUP_BY_FOLDER_NAME, conf_file_path)
+    else:
+        return None  # Handle other cases or errors if needed
+
+
+def _print_features(obj, obj_class):
```

Comment:
not used?

Reply:
This PR is based on top of https://github.com/airbnb/chronon/pull/819. Some carry over. I will rebase once the 819 is merged